pygments/lexers/c_cpp.py

```diff
@@ -118,19 +133,25 @@ class CFamilyLexer(RegexLexer):
             include('keywords'),
             # functions
             (r'(' + _namespaced_ident + r'(?:[&*\s])+)'  # return arguments
+             r'((?:(?://.*?\n)|(?:/[*][\w\W]*?[*]/)|\s*?)*)'    # possible comments
```

Comment:
This regex has catastrophic backtracking. Try matching input like

```
int
                                                                                    // lots of spaces
+++ // whoops, invalid
```


It should take a loooong time. See `CONTRIBUTING.md` for more explanation. Then change this to

```
r'((?:((?://.*?\n)|(?:/[*](.|\n)*?[*]/))\s*)*)'
```

Also, please use a variable to avoid repeating this regex below.

Reply:
+1 for using a variable. Additionally, @lambda-karlculus, is there a reason for not using the existing regular expressions for comments?
https://github.com/pygments/pygments/blob/81f1669f43cf24a33e720317b3f9b60991636d4e/pygments/lexers/c_cpp.py#L78-L79